mne/preprocessing/artifact_detection.py

```diff
@@ -4,12 +4,113 @@
 from ..transforms import (quat_to_rot, _average_quats, _angle_between_quats,
                           apply_trans, _quat_to_affine)
+from ..filter import filter_data
 from .. import Transform
-from ..utils import (_mask_to_onsets_offsets, logger)
+from ..utils import (_mask_to_onsets_offsets, logger, verbose)
+
+
+@verbose
+def annotate_muscle_zscore(raw, threshold=4, ch_type=None, min_length_good=0.1,
+                           filter_freq=(110, 140), n_jobs=1, verbose=None):
+    """Create annotations for segments that likely contain muscle artifacts.
+
+    Detects data segments containing activity in the frequency range given by
+    ``filter_freq`` whose envelope magnitude exceeds the specified z-score
+    threshold, when summed across channels and divided by ``sqrt(n_channels)``.
+    False-positive transient peaks are prevented by low-pass filtering the
+    resulting z-score time series at 4 Hz. Only operates on a single channel
+    type, if ``ch_type`` is ``None`` it will select the first type in the list
+    ``mag``, ``grad``, ``eeg``.
+    See :footcite:`Muthukumaraswamy2013` for background on choosing
+    ``filter_freq`` and ``threshold``.
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        Data to estimate segments with muscle artifacts.
+    threshold : float
+        The threshold in z-scores for marking segments as containing muscle
+        activity artifacts.
+    ch_type : 'mag' | 'grad' | 'eeg' | None
+        The type of sensors to use. If ``None`` it will take the first type in
+        ``mag``, ``grad``, ``eeg``.
+    min_length_good : float | None
+        The shortest allowed duration of "good data" (in seconds) between
+        adjacent annotations; shorter segments will be incorporated into the
+        surrounding annotations.``None`` is equivalent to ``0``.
+        Default is ``0.1``.
+    filter_freq : array-like, shape (2,)
+        The lower and upper frequencies of the band-pass filter.
+        Default is ``(110, 140)``.
+    %(n_jobs)s
+    %(verbose)s
+
+    Returns
+    -------
+    annot : mne.Annotations
+        Periods with muscle artifacts annotated as BAD_muscle.
+    scores_muscle : array
+        Z-score values averaged across channels for each sample.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    from scipy.stats import zscore
+    from scipy.ndimage.measurements import label
+
+    raw_copy = raw.copy()
+
+    if ch_type is None:
+        raw_ch_type = raw_copy.get_channel_types()
+        if 'mag' in raw_ch_type:
+            ch_type = 'mag'
+        elif 'grad' in raw_ch_type:
+            ch_type = 'grad'
+        elif 'eeg' in raw_ch_type:
+            ch_type = 'eeg'
+        logger.info('Using %s sensors for muscle artifact detection'
+                    % (ch_type))
+
+    if ch_type in ('mag', 'grad'):
+        raw_copy.pick_types(meg=ch_type, ref_meg=False)
+    elif ch_type == 'eeg':
+        raw_copy.pick_types(meg=False, eeg=True)
```

Comment:
there should be an `else:` raising `ValueError` - saying that annotate_muscle_zscore accepts only mag, grad or eeg ch_type.

Reply:
@mmagnuski I have never seen you make a commit suggestion, only comments. Do you know how to do them?

Reply:
@AdoNunes using "suggestion mode" in PR comments and reviews is optional; I often do it when I think it will be faster than explaining my suggestion in words, but nobody is required to do it that way.

Reply:
@AdoNunes: Yes, I do :)

Reply:
indeed if the data does not contain any eeg or meg (eg fnirs, seeg, ecog) this code will break. It needs an else with an exception raised. @AdoNunes if you need to move on, it can wait and probably someone can take over.

Reply:
@agramfort  I can change it, I just thought that without the else it could be used with other sensors, like for detecting blinks wit ecog filtered at low frequencies, or fnirs.

Reply:
@AdoNunes : currently if you did `ch_type='eog'` all the data would be used, while the user would be under impression that correct channel types are picked. If we want to support other types, why not turn the `ch_type` to dict (for example `ch_type = {'meg': False, ch_type: True}`) in the non-meg case and then do `raw_copy.pick_types(**ch_type)`? Any resulting errors should be raised by `pick_types` then.

Reply:
Sounds good, it leaves the door open to using other sensor types in case it is needed